docs/slicing_and_indices.md

```diff
@@ -0,0 +1,186 @@
+# Slicing, Indexing and Data retrieval
+
+## Converting positions to indices
+
+A position is converted to an index in several steps:
+
+1. Scaling if approriate. This applies to Sampled and Range Dimensions if they contain units. **Note:** Incompatible units will always lead to an error!
+2. Converting to an index depending on a) the dimension type and b) the *PositionMatching* rule.
+
+Five possible rules of *PositionMatching* (we always assuming ascending order in the dimension):
+
+1. **Less**: the index points to the last entry that is less than position.
+2. **LessOrEqual**: the index points to the last entry that is less or equal to position.
+3. **Equal**: the index points to the entry that exatly matches position.
+4. **GreaterOrEqual**: the intex points to the first entry that is greater or equal to position.
+5. **Greater**: the index points to the first entry that is greater than position.
+
+For the different DimensionTypes slightly different rules apply:
+
+### SampledDimension
+
+Assume a SampledDimension with *sampling_interval* = 1 and *offset* = -1
+x = [-1, 0., 1, 2, 3, 4, 5, ... n]
+
```

Comment:
```suggestion
The table below shows three example positions `p = -2, -1, 0` combined with each `PositionMatching` rule and the expected result.
```

Maybe also swap the `index` and `PositionRule` columns or even put `index` last.  Since `p` and `PositionRule` are the _input_ to this function, so to speak, the `index` is the _output_, and `validity` is important information about the behaviour and output.

Reply:
I'd add a similar description of what the tables show to the following ones.  Or maybe a single description at the top for all.